source/cppfront.cpp

```diff
@@ -5451,8 +5473,16 @@ class cppfront
                     decl->get_decl_if_type_scope_object_name_before_a_base_type(*decl->name())
                     || decl->has_name("this")
                     ;
+
                 if (emit_as_base)
                 {
+                    if (started_body) {
+                        errors.emplace_back(
+                            stmt->position(),
+                            "(temporary alpha limitation) expected only objects before 'this' members"
```

Comment:
This takes care of <https://github.com/hsutter/cppfront/issues/471#issuecomment-1793940564>.